plugins/modules/opensearch.py

```diff
@@ -0,0 +1,1194 @@
+    domain_name: "my-domain"
+    engine_version: OpenSearch_1.1
+    cluster_config:
+      instance_type: "m3.medium.search"
+      instance_count: 2
+      dedicated_master: true
+      zone_awareness: true
+      availability_zone_count: 2
+      dedicated_master_instance_type: "t2.micro.search"
+      dedicated_master_instance_count: 2
+      warm_enabled: true
+      warm_type: "ultrawarm1.medium.search"
+      warm_count: 1
+      cold_storage_options:
+        enabled: false
+    ebs_options:
+      enabled: true
+      volume_type: "io1"
+      volume_size: 10
+      iops: 1000
+    vpc_options:
+      subnets:
+        - "subnet-e537d64a"
+        - "subnet-e537d64b"
+      security_groups:
+        - "sg-dd2f13cb"
+        - "sg-dd2f13cc"
+    snapshot_options:
+      automated_snapshot_start_hour: 13
+    access_policies: "{{ lookup('file', 'policy.json') | from_json }}"
+    encryption_at_rest_options:
+      enabled: false
+    node_to_node_encryption_options:
+      enabled: false
+    auto_tune_options:
+      enabled: true
+      maintenance_schedules:
+      - start_at: "2025-01-12"
+        duration:
+          value: 1
+          unit: "HOURS"
+        cron_expression_for_recurrence: "cron(0 12 * * ? *)"
+      - start_at: "2032-01-12"
+        duration:
+          value: 2
+          unit: "HOURS"
+        cron_expression_for_recurrence: "cron(0 12 * * ? *)"
+    tags:
+      Environment: Development
+      Application: Search
+    wait: true
+
+"""
+
+from copy import deepcopy
+import json
+import time
+
+try:
+    import botocore
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+from ansible.module_utils.six import string_types
+
+# import module snippets
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import (
+    camel_dict_to_snake_dict,
+    AWSRetry,
+    compare_aws_tags,
+)
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import (
+    boto3_tag_list_to_ansible_dict,
+    ansible_dict_to_boto3_tag_list,
+)
+
+
+def get_domain(client, module, domain_name):
+    try:
+        response = client.describe_domain(DomainName=domain_name)
+    except client.exceptions.ResourceNotFoundException:
+        return None
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(e, msg="Couldn't get domain {0}".format(domain_name))
+    return response["DomainStatus"]
+
+
+def get_domain_config(client, module, domain_name):
+    try:
+        response = client.describe_domain_config(DomainName=domain_name)
+    except client.exceptions.ResourceNotFoundException:
+        return None
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(e, msg="Couldn't get domain {0}".format(domain_name))
+    return response["DomainConfig"]
+
+
+def domain_to_facts(client, module, domain):
```

Comment:
Consider also moving this one here https://github.com/ansible-collections/community.aws/tree/main/plugins/module_utils inside the opensearch.py dedicated file. I would personally change the name of the method since it only returns the associated tags.